spec/unit/facter/network_nexthop_ip_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'spec_helper'
+
+describe Facter.fact(:network_nexthop_ip) do
+  before { Facter.clear }
+  after { Facter.clear }
+
+  describe 'network_nexthop_ip' do
+    context' returns gateway' do
+      before do
+        allow(Facter.fact(:kernel)).to receive(:value).and_return(:linux)
+        allow(Facter::Util::Resolution).to receive(:which).with('ip').and_return('/usr/bin/ip')
+      end
+      let(:facts) { { kernel: :linux } }
+      it { expect(Facter::Util::Resolution).to receive(:exec).with('ip route show 0/0').and_return('default via 192.168.178.1 dev eth0') }
```

Comment:
This looks wrong. Typically you have this as part of a single `it`. Now you have a test that expects something to be run, but nothing actually calls the fact so it isn't run. In the next line you have a test to call the fact, but nothing mocks the exec so it fails.